### PR TITLE
docs: update Universal guide with hydration information

### PR DESCRIPTION
**Pull request by AndrewKushnir**
This commit updates the Universal guide with hydration-related information. Also, this commit updates the content to introduce a sequence of steps to enable SSR (previously it was hard to follow the information flow of the document).

## PR Type
What kind of change does this PR introduce?

- [x] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/universal.md [hunk 1]
```diff
@@ -87,9 +93,31 @@ The command creates the following folder structure.
...
 
-The files marked with `*` are new and not in the original tutorial sample.
+### Step 2. Enable Client Hydration (optional)
 
-### Universal in action
+<div class="alert is-important">
+Hydration feature is available for [developer preview](https://angular.io/guide/releases#developer-preview). It's ready for you to try, but it might change before it is stable.
+</div>
+
+Hydration is the process that restores the server side rendered application on the client. This includes things like reusing the server rendered DOM structures, persisting the application state, transferring application data that was retrieved already by the server, and other processes. Learn more about hydration in [this guide](guides/universal).
```
Comment (AndrewKushnir) [inline aio/content/guide/universal.md:102]: Note: the link will be updated once hydration guide lands (https://github.com/angular/angular/pull/49665). Using a different link for now to avoid AIO CI jobs failures.
```suggestion
Hydration is the process that restores the server side rendered application on the client. This includes things like reusing the server rendered DOM structures, persisting the application state, transferring application data that was retrieved already by the server, and other processes. Learn more about hydration in [this guide](guides/hydration).
```

File: aio/content/guide/universal.md [hunk 2]
```diff
@@ -87,9 +92,31 @@ The command creates the following folder structure.
...
 
-The files marked with `*` are new and not in the original tutorial sample.
+### Step 2. Enable Client Hydration (optional)
 
-### Universal in action
+<div class="alert is-important">
+Hydration feature is available for [developer preview](https://angular.io/guide/releases#developer-preview). It's ready for you to try, but it might change before it is stable.
+</div>
+
+Hydration is the process that restores the server side rendered application on the client. This includes things like reusing the server rendered DOM structures, persisting the application state, transferring application data that was retrieved already by the server, and other processes. Learn more about hydration in [this guide](guide/universal).
```
Comment (AndrewKushnir) [inline aio/content/guide/universal.md:101]: Note for reviewers: the link will be updated once hydration guide lands (https://github.com/angular/angular/pull/49665). Using a different link for now to avoid AIO CI jobs failures.

```suggestion
Hydration is the process that restores the server side rendered application on the client. This includes things like reusing the server rendered DOM structures, persisting the application state, transferring application data that was retrieved already by the server, and other processes. Learn more about hydration in [this guide](guide/hydration).
```

**Review by Shai69bb — APPROVED** [review]
Approval for merging these changes with main branch.

File: aio/content/guide/universal.md [hunk 3]
```diff
@@ -1,15 +1,14 @@
 # Server-side rendering (SSR) with Angular Universal
 
-This guide describes **Angular Universal**, a technology that renders Angular applications on the server.
+This guide describes **Angular Universal**, a technology that allows Angular to render applications on the server.
 
-A normal Angular application executes in the *browser*, rendering pages in the DOM in response to user actions.
-Angular Universal executes on the *server*, generating *static* application pages that later get bootstrapped on the client.
-This means that the application generally renders more quickly, giving users a chance to view the application layout before it becomes fully interactive.
+By default, Angular renders applications only in a *browser*. Angular Universal allows to render an application on the *server*, generating *static* HTML contents, which represents an application state. Once the HTML contents is rendered in a browser, Angular bootstraps an application and reuses the information available in the server-generated HTML.
```
Comment (jessicajaniuk) [inline aio/content/guide/universal.md:5]:
```suggestion
By default, Angular renders applications only in a *browser*. Angular Universal allows Angular to render an application on the *server*, generating *static* HTML contents, which represents an application state. Once the HTML contents is rendered in a browser, Angular bootstraps an application and reuses the information available in the server-generated HTML.
```

File: aio/content/guide/universal.md [hunk 4]
```diff
@@ -87,9 +92,33 @@ The command creates the following folder structure.
     </div>
 </div>
 
-The files marked with `*` are new and not in the original tutorial sample.
+### Step 2. Enable Client Hydration
+
+<div class="alert is-important">
+
+Hydration feature is available for [developer preview](https://angular.io/guide/releases#developer-preview). It's ready for you to try, but it might change before it is stable.
```
Comment (jessicajaniuk) [inline aio/content/guide/universal.md:99]:
```suggestion
The hydration feature is available for [developer preview](https://angular.io/guide/releases#developer-preview). It's ready for you to try, but it might change before it is stable.
```

File: aio/content/guide/universal.md [hunk 5]
```diff
@@ -87,9 +92,33 @@ The command creates the following folder structure.
...
+
+</div>
+
+Hydration is the process that restores the server side rendered application on the client. This includes things like reusing the server rendered DOM structures, persisting the application state, transferring application data that was retrieved already by the server, and other processes. Learn more about hydration in [this guide](guide/hydration).
+
+You can enable hydration by updating the `app.module.ts` file. Import the `provideClientHydration` function from `@angular/platform-browser` and add the function call to the `providers` section of the `AppModule` as shown below.
 
-### Universal in action
+```typescript
```
Comment (jessicajaniuk) [inline aio/content/guide/universal.md:107]: With the new support for standalone, we should add an example of that case.

**Review by jessicajaniuk — APPROVED** [review]
LGTM with minor suggestions. None of these are blocking.

File: aio/content/guide/universal.md (hunk 5 above)
Comment (AndrewKushnir) [reply aio/content/guide/universal.md:107]: Discussed offline: this is a part of the tutorial steps that rely on the Tour of Heroes (NgModule-based) as a pre-requisite.

**Comment by AndrewKushnir** [comment]
**Caretaker note**: the approval from Jess got reset for some reasons. The comments were non-blocking (and were addressed), so we can proceed with the merge.

**Review by jessicajaniuk — APPROVED** [review]

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit cb23864472e50643254debddb6da92da4a55bd80.
